### PR TITLE
feat(HMS-1773): Add IPv4 to GCP instance desc

### PR DESCRIPTION
**Pull request by adiabramovitch**
This PR partially depends on #468, in order to see that the instances description is available, still WIP. 
Wanted to hear your thoughts regarding the functions I have added to GCP's interface.   
It seems like there is only internal DNS available according to google docs, so currently I only added IPV4 info.

### CODE REVIEW

**Review by lzap — CHANGES_REQUESTED** [review]
That was my conclusion too when I was adding public IP to insights-core, only IPv4 is available at this point.

Looks great, few nitpicks here and there but it does the job nicely. The ID being string is ugly but as long as we want to reuse the type for other clouds I think this is fine. It is only informative at this point, so not problem.

File: internal/clients/http/gcp/gcp_client.go [hunk 1]
```diff
@@ -163,41 +160,87 @@ func (c *gcpClient) InsertInstances(ctx context.Context, params *clients.GCPInst
...
+	projectId := c.auth.String()
+
+	instance, err := client.Get(ctx, &computepb.GetInstanceRequest{Instance: id, Project: projectId})
+	if err != nil {
+		return nil, fmt.Errorf("unable to get instance: %w", err)
+	}
+	instanceId := strconv.FormatUint(instance.GetId(), 10)
+	instanceDesc := clients.InstanceDescription{ID: instanceId}
+	for _, n := range instance.NetworkInterfaces {
+		instanceDesc.PublicIPv4 = *n.AccessConfigs[0].NatIP
```
[the remark this answers is not in the document]
Comment (adiabramovitch) [reply internal/clients/http/gcp/gcp_client.go:243]: right
Can we assume we only have one NetworkInterface for now? and do something like this by verifying it was allocated: 
```golang
	if len(instance.NetworkInterfaces) > 0 {
		inter := instance.NetworkInterfaces[0]
		instanceDesc.ID = instanceId
		if len(inter.AccessConfigs) > 0 {
			instanceDesc.PublicIPv4 = *inter.AccessConfigs[0].NatIP
		}
	}
```

File: internal/clients/http/gcp/gcp_client.go [hunk 2]
```diff
@@ -164,41 +162,90 @@ func (c *gcpClient) InsertInstances(ctx context.Context, params *clients.GCPInst
...
+
+	instance, err := client.Get(ctx, &computepb.GetInstanceRequest{Instance: id, Project: projectId})
+	if err != nil {
+		return nil, fmt.Errorf("unable to get instance: %w", err)
+	}
+	instanceId := strconv.FormatUint(instance.GetId(), 10)
+	instanceDesc := clients.InstanceDescription{ID: instanceId}
+	for _, n := range instance.NetworkInterfaces {
+		if len(n.AccessConfigs) > 0 {
+			instanceDesc.PublicIPv4 = *n.AccessConfigs[0].NatIP
```
Comment (lzap) [inline internal/clients/http/gcp/gcp_client.go:246]: I still have the initial concern, you dereference something that might be `nil`. This needs to be:

```go
if n.AccessConfigs[0] != nil {
  instanceDesc.PublicIPv4 = *n.AccessConfigs[0].NatIP
}
```

**Review by lzap — CHANGES_REQUESTED** [review]
One dereferencing comment, please rebase (not caused by the logger refactoring I believe).

**Review by lzap — APPROVED** [review]
Perfect work, thank you.

**Pull request merged**